test/unit/BondingManager.js

```diff
@@ -513,6 +513,24 @@ describe("BondingManager", () => {
                 )
             })
 
+            it("should set delegate for different address", async () => {
```

Comment:
I think it would be clearer if the test cases specific to `bondForWithHint()` are grouped under their own describe block. This would make it easier to review testing of functionality that is specific to `bondForWithHint()` (i.e. the new cases added thus far here). WDYT?